### PR TITLE
Disable NSToolbar Tabs

### LINKED ISSUE #23380: [BUG] iPadOS 18 Forced TabView
### Description

Upon iPadOS 18's Dev Beta's release. They introduced a new way to navigate through various pages, "TabView". Although it can be useful, you can't really get rid of it in .NET MAUI.

So instead you'll experience weird behaviors that aren't intended like so:
image

It looks like the iPadOS version of the app assumes there is a tabview available but I can't really say that for sure.

### Steps to Reproduce

Step 1) Create a new MAUI Project.
Step 2 [Optional]) Set the " Shell.NavBarIsVisible="False" " flag in the MainPage.
Step 3) Run it under an iPad Simulator on iPadOS 18.

If you did all of the above you'll see this TabView showing up out of nowhere with no way to get rid of it.

### Link to public reproduction project repository

https://github.com/imadofficial/TabViewIssue

### Version with bug

8.0.21 SR4.1

### Is this a regression from previous behavior?

No, this is something new

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iPadOS 18 (iOS 18 isn't affected)

### Did you find any workaround?

I wasn't able to find a workaround for this.

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by PureWeen**
### Description of Change

Backport of this code from main
https://github.com/dotnet/maui/blob/76a76fbaef3f1a63fc2b554f9deb7eb3897c4e98/src/Core/src/Platform/iOS/TabbedViewExtensions.cs#L11

### Issues Fixed
Fixes #23380

### CODE REVIEW

File: src/Core/src/Platform/iOS/TabbedViewExtensions.cs [hunk 1]
```diff
@@ -9,6 +9,22 @@ namespace Microsoft.Maui.Platform
 {
 	internal static class TabbedViewExtensions
 	{
+		internal static void DisableiOS18ToolbarTabs(
+			this UITabBarController tabBarController)
+		{
+
+			bool is18 = OperatingSystem.IsMacCatalystVersionAtLeast(18);
```
Comment (pictos) [inline src/Core/src/Platform/iOS/TabbedViewExtensions.cs:16]: looks like this isn't used anywhere

Comment (PureWeen) [reply src/Core/src/Platform/iOS/TabbedViewExtensions.cs:16]: Whoops

Yea, this command is running incorrectly for me locally and I accidently left this in :-)

**Comment by PureWeen** [comment]
This one is on hold until we merge

https://github.com/dotnet/maui/pull/25140

File: src/Core/src/Platform/iOS/TabbedViewExtensions.cs [hunk 2]
```diff
@@ -9,6 +9,20 @@ namespace Microsoft.Maui.Platform
...
+		internal static void DisableiOS18ToolbarTabs(
+			this UITabBarController tabBarController)
+		{
+			// Should apply to iOS and Catalyst
+            if (OperatingSystem.IsMacCatalystVersionAtLeast(18,0,-1) || //https://github.com/xamarin/xamarin-macios/issues/21390
+				OperatingSystem.IsIOSVersionAtLeast(18,0))
+            {
+				tabBarController.TraitOverrides.HorizontalSizeClass = UIUserInterfaceSizeClass.Compact;
+                tabBarController.Mode = UITabBarControllerMode.TabSidebar;
+                tabBarController.Sidebar.Hidden = true;
```
Comment (jsuarezruiz) [inline src/Core/src/Platform/iOS/TabbedViewExtensions.cs:21]:
```
D:\a\_work\1\s\src\Core\src\Platform\iOS\TabbedViewExtensions.cs(20,34): error CS1061: 'UITabBarController' does not contain a definition for 'Mode' and no accessible extension method 'Mode' accepting a first argument of type 'UITabBarController' could be found (are you missing a using directive or an assembly reference?) [D:\a\_work\1\s\src\Core\src\Core.csproj::TargetFramework=net8.0-ios]
D:\a\_work\1\s\src\Core\src\Platform\iOS\TabbedViewExtensions.cs(20,41): error CS0103: The name 'UITabBarControllerMode' does not exist in the current context [D:\a\_work\1\s\src\Core\src\Core.csproj::TargetFramework=net8.0-ios]
D:\a\_work\1\s\src\Core\src\Platform\iOS\TabbedViewExtensions.cs(21,34): error CS1061: 'UITabBarController' does not contain a definition for 'Sidebar' and no accessible extension method 'Sidebar' accepting a first argument of type 'UITabBarController' could be found (are you missing a using directive or an assembly reference?) [D:\a\_work\1\s\src\Core\src\Core.csproj::TargetFramework=net8.0-ios]
D:\a\_work\1\s\src\Core\src\Platform\iOS\TabbedViewExtensions.cs(22,34): error CS1061: 'UITabBarController' does not contain a definition for 'TabBarHidden' and no accessible extension method 'TabBarHidden' accepting a first argument of type 'UITabBarController' could be found (are you missing a using directive or an assembly reference?) [D:\a\_work\1\s\src\Core\src\Core.csproj::TargetFramework=net8.0-ios]
    36 Warning(s)
    16 Error(s)
```

Comment (drasticactions) [reply src/Core/src/Platform/iOS/TabbedViewExtensions.cs:21]: That's expected, https://github.com/dotnet/maui/pull/25140 needs to be merged first to get the newest SDK with the bindings.

**Review by StephaneDelcroix — APPROVED** [review]

**Review by Redth — APPROVED** [review]

**Review by jsuarezruiz — APPROVED** [review]

File: src/Controls/src/Core/Compatibility/Handlers/Shell/iOS/ShellItemRenderer.cs [hunk 3]
```diff
@@ -438,7 +439,14 @@ void UpdateTabBarHidden()
...
 				return;
 
-			TabBar.Hidden = !ShellItemController.ShowTabs;
+			if (OperatingSystemMacCatalyst18Workaround.IsMacCatalystVersionAtLeast18() || OperatingSystem.IsIOSVersionAtLeast(18))
+			{
+				TabBarHidden = !ShellItemController.ShowTabs;
+			}
+			else
+			{
+				TabBar.Hidden = !ShellItemController.ShowTabs;
```
Comment (jsuarezruiz) [inline src/Controls/src/Core/Compatibility/Handlers/Shell/iOS/ShellItemRenderer.cs:448]:
```
D:\a\_work\1\s\src\Controls\src\Core\Compatibility\Handlers\Shell\iOS\ShellItemRenderer.cs(444,5): error CS0103: The name 'TabBarHidden' does not exist in the current context [D:\a\_work\1\s\src\Controls\src\Core\Controls.Core.csproj::TargetFramework=net8.0-maccatalyst17.0]
D:\a\_work\1\s\src\Controls\src\Core\Compatibility\Handlers\Shell\iOS\ShellItemRenderer.cs(444,5): error CS0103: The name 'TabBarHidden' does not exist in the current context [D:\a\_work\1\s\src\Controls\src\Core\Controls.Core.csproj::TargetFramework=net8.0-ios17.0]
    36 Warning(s)
    2 Error(s)

```

**Review by jsuarezruiz — CHANGES_REQUESTED** [review]

File: src/Controls/src/Core/Compatibility/Handlers/Shell/iOS/ShellItemRenderer.cs (hunk 3 above)
Comment (jsuarezruiz) [reply src/Controls/src/Core/Compatibility/Handlers/Shell/iOS/ShellItemRenderer.cs:448]: Requires` ios18.0`.

**Comment by jsuarezruiz** [comment]
Fixing CI, the changes works as expected.

**Review by jsuarezruiz — APPROVED** [review]

**Pull request merged**